odxtools/database.py

```diff
@@ -39,20 +37,23 @@ def __init__(self,
         self._comparam_subsets = NamedItemList[ComparamSubset]()
         self._comparam_specs = NamedItemList[ComparamSpec]()
 
-    def add_pdx_file(self, pdx_file_name: str) -> None:
-        pdx_zip = ZipFile(pdx_file_name)
-
-        for zip_member in pdx_zip.namelist():
+    def add_pdx_file(self, pdx_file: Union[str, PathLike[str], IO[bytes], ZipFile]) -> None:
+        """Add PDX file to database.
+        Either pass the path to the file, an IO with the file content or a ZipFile object.
+        """
+        if not isinstance(pdx_file, ZipFile):
+            pdx_file = ZipFile(pdx_file)
```

Comment:
```suggestion
        if isinstance(pdx_file, ZipFile):
            pdx_zip = pdx_file
        else:
            pdx_zip = ZipFile(pdx_file)
```

with this, the rest of the function can be left untouched...